File: space-game/Core.Tests/Data/AngleTestsData.vb
Imports Core.Classes
Imports NUnit.Framework

Namespace Data
    Public Class AngleTestsData

        Public Shared Function Equals_WhenComparing_ReturnsExpected_Data() As List(Of TestCaseData)
            Dim nullAngle As Angle = Nothing
            Dim a = Angle.FromDegrees(123.456)
            Dim copyOfA = a
            Dim notSame = Angle.FromDegrees(12)

            Return New List(Of TestCaseData) From {
                New TestCaseData(a, nullAngle, False),
                New TestCaseData(a, copyOfA, True),
                New TestCaseData(a, notSame, False)
            }
        End Function

    End Class
End Namespace

File: space-game/Core.Tests/Data/AngleTestsData.cs
using Core.Classes;
using NUnit.Framework;
using System.Collections.Generic;

namespace Core.Tests.Data
{
    public class AngleTestsData
    {
        public static List<TestCaseData> Equals_WhenComparing_ReturnsExpected_Data()
        {
            Angle nullAngle = null;
            Angle a = Angle.FromDegrees(123.456);
            Angle copyOfA = a;
            Angle notSame = Angle.FromDegrees(12);

            return new List<TestCaseData> {
                new TestCaseData(a, nullAngle, false),
                new TestCaseData(a, copyOfA, true),
                new TestCaseData(a, notSame, false)
            };
        }
    }
}
